[CHI-137] add floating label support to inputs

dani-cl-madrid opened the pull request:
Developed CSS only floating label. It supports all major browsers but Microsoft Edge. It even support Internet Explorer 10 and 11.
Pending develop a JavaScript solution for Microsoft Edge.

jllr commented:
@dani-cl-madrid as it does not work in Edge, we should move towards a JS solution. @mattnickles thoughts?

dani-cl-madrid commented:
Removed pure CSS solution.
Developed JS solution based on -active class on the label element.
Fixed input paddings.
Reset input font-family to the one set to body instead of sans-serif defined by normalized.css.
New JS Util.merge function for configuration.
New JS util.addArraySupportToFactory for configuration. Now every component factory method can receive an array of elements and will instantiate all of them. Good for getElementsByClassName instead of getElementById.
